sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -2434,829 +2447,799 @@ class CollationSQLExpressionsSuite
     )
   }
 
-  test("min_by supports collation") {
-    val collation = "UNICODE"
-    val query = s"SELECT min_by(x, y) FROM VALUES ('a', 10), ('b', 50), ('c', 20) AS tab(x, y);"
+  // common method for subsequent tests verifying various SQL expressions with collations
+  private def testCollationSqlExpressionCommon(
+      query: String,
+      collation: String,
+      result: Seq[Row],
+      expectedTypes: Seq[DataType]): Unit = {
     withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
-      checkAnswer(
-        sql(query),
-        Seq(
+      // check result correctness
+      checkAnswer(sql(query), result)
+      // check result row data types
+      for (i <- 0 until expectedTypes.length)
+        assert(sql(query).schema(i).dataType == expectedTypes(i))
+    }
+  }
+
+  test("min_by supports collation") {
+    testAdditionalCollations.foreach { collation =>
+      testCollationSqlExpressionCommon(
+        query = "SELECT min_by(x, y) FROM VALUES ('a', 10), ('b', 50), ('c', 20) AS tab(x, y);",
+        collation,
+        result = Seq(
           Row("a")
+        ),
+        expectedTypes = Seq(
+          StringType(collation)
         )
       )
-      // check result row data type
-      val dataType = StringType(collation)
-      assert(sql(query).schema.head.dataType == dataType)
     }
   }
```

Comment:
Let's simplify the common use-case, and add an overloaded versions `testCollationSqlExpressionCommon`:
```scala
  private def testCollationSqlExpressionCommon(
      query: String,
      collation: String,
      result: Row,
      expectedType: DataType): Unit = {
    testCollationSqlExpressionCommon(query, collation, Seq(result), Seq(expectedType))
  }
```
like
```scala
  test("min_by supports collation") {
    testAdditionalCollations.foreach { collation =>
      testCollationSqlExpressionCommon(
        query = "SELECT min_by(x, y) FROM VALUES ('a', 10), ('b', 50), ('c', 20) AS tab(x, y);",
        collation,
        result = Row("a"),
        expectedType = StringType(collation)
      )
    }
  }
```

Reply:
Good suggestion, I've added an overloaded version of the method and used the simpler version throughout the test suite where possible, thanks!

Reply:
@MaxGekk, I have addressed all the comments so far, the tests should be much more readable now (thanks for all the great suggestions!) - please let me know if you see any further issues or the change can be merged now, thanks!